watchtower-plugin/src/retrier.rs

```diff
@@ -16,174 +16,248 @@ use crate::wt_client::WTClient;
-                                // can happen.
-                                log::info!(
-                                    "Some data was missed while retrying {}. Adding it back",
-                                    tower_id
-                                );
-                                for locator in retrier.pending_appointments.lock().unwrap().drain()
-                                {
-                                    state.unreachable_towers.send((tower_id, locator)).unwrap();
-                                }
-                            } else {
-                                log::info!("Retry strategy succeeded for {}", tower_id);
-                                state.set_tower_status(tower_id, crate::TowerStatus::Reachable);
-                            }
-                        }
-                        Err(e) => {
-                            log::warn!("Retry strategy gave up for {}. {}", tower_id, e);
-                            // Notice we'll end up here after a permanent error. That is, either after finishing the backoff strategy
-                            // unsuccessfully or by manually raising such an error (like when facing a tower misbehavior)
-                            if let Some(tower) = state.towers.get_mut(&tower_id) {
-                                if tower.status.is_temporary_unreachable() {
-                                    log::warn!("Setting {} as unreachable", tower_id);
-                                    state.set_tower_status(
-                                        tower_id,
-                                        crate::TowerStatus::Unreachable,
-                                    );
-                                }
-                            } else {
-                                log::info!("Skipping retrying abandoned tower {}", tower_id);
-                            }
+                Err(TryRecvError::Empty) => {
+                    // Keep only running retriers and retriers ready to be started/re-started.
+                    // This will remove failed ones and ones finished successfully and have no pending appointments.
+                    //
+                    // Note that a failed retrier could have received some new appointments to retry. In this case, we don't try to send
+                    // them because we know that that tower is unreachable. We most likely received these new appointments because the tower
+                    // was marked as temporary unreachable during the retrial process so the sender didn't know that the tower is unreachable.
+                    // But this can also happen if the user manually retries the tower just right after (within 1 second time frame) it being
+                    // marked as unreachable. The user needs to wait some time and try again when we have remove the retrier for that tower.
```

Comment:
I think this issue can be fixed by flagging the tower as unreachable here, instead of as a result of the backoff strategy, that is:

The retrier is flagged as failed by the backoff failure, then it gets here, its drained as failed and the corresponding tower is flagged. In that way the user won't be able to manually retry until the retrier is actually wiped, given the tower being unreachable is a requirement for that.